rocmPackages: 6.4.3 -> 7.0.2

LunNova opened the pull request:
For #462424

Bumping to 7.0.x first, 7.1.x will be separate.

Upstream patches raised:

- https://github.com/ROCm/rocm-systems/pull/2423
- https://github.com/ROCm/rocm-systems/pull/2424
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 469378`
Commit: `c393d4f2a452ac047c10e7fceba4e3f646e51fb4`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 126 packages built:</summary>
  <ul>
    <li>adaptivecpp</li>
    <li>adaptivecppWithCuda</li>
    <li>adaptivecppWithRocm</li>
    <li>blender-hip</li>
    <li>btop-rocm</li>
    <li>llama-cpp-rocm</li>
    <li>magma (magma-hip)</li>
    <li>magma-cuda-static</li>
    <li>magma-cuda-static.test</li>
    <li>magma.test (magma-hip.test)</li>
    <li>ollama-rocm</li>
    <li>opensplat</li>
    <li>opensplatWithCuda</li>
    <li>opensplatWithRocm</li>
    <li>python313Packages.torchWithCuda</li>
    <li>python313Packages.torchWithCuda.cxxdev</li>
    <li>python313Packages.torchWithCuda.dev</li>
    <li>python313Packages.torchWithCuda.dist</li>
    <li>python313Packages.torchWithCuda.lib</li>
    <li>python313Packages.torchWithRocm</li>
    <li>python313Packages.torchWithRocm.cxxdev</li>
    <li>python313Packages.torchWithRocm.dev</li>
    <li>python313Packages.torchWithRocm.dist</li>
    <li>python313Packages.torchWithRocm.lib</li>
    <li>python314Packages.torchWithCuda</li>
    <li>python314Packages.torchWithCuda.cxxdev</li>
    <li>python314Packages.torchWithCuda.dev</li>
    <li>python314Packages.torchWithCuda.dist</li>
    <li>python314Packages.torchWithCuda.lib</li>
    <li>python314Packages.torchWithRocm</li>
    <li>python314Packages.torchWithRocm.cxxdev</li>
    <li>python314Packages.torchWithRocm.dev</li>
    <li>python314Packages.torchWithRocm.dist</li>
    <li>python314Packages.torchWithRocm.lib</li>
    <li>rocmPackages.amdsmi</li>
    <li>rocmPackages.aotriton</li>
    <li>rocmPackages.aotriton.debug</li>
    <li>rocmPackages.aqlprofile</li>
    <li>rocmPackages.ck4inductor</li>
    <li>rocmPackages.ck4inductor.dist</li>
    <li>rocmPackages.clang (rocmPackages.llvm.clang, rocmPackages.llvm.rocm-toolchain, rocmPackages.rocm-toolchain)</li>
    <li>rocmPackages.clr</li>
    <li>rocmPackages.clr.debug</li>
    <li>rocmPackages.clr.icd</li>
    <li>rocmPackages.composable_kernel</li>
    <li>rocmPackages.half</li>
    <li>rocmPackages.hip-common</li>
    <li>rocmPackages.hipblas</li>
    <li>rocmPackages.hipblas-common</li>
    <li>rocmPackages.hipblaslt</li>
    <li>rocmPackages.hipblaslt.benchmark</li>
    <li>rocmPackages.hipcc</li>
    <li>rocmPackages.hipcub</li>
    <li>rocmPackages.hipfft</li>
    <li>rocmPackages.hipfort</li>
    <li>rocmPackages.hipify</li>
    <li>rocmPackages.hiprand</li>
    <li>rocmPackages.hiprt</li>
    <li>rocmPackages.hipsolver</li>
    <li>rocmPackages.hipsparse</li>
    <li>rocmPackages.hsakmt (rocmPackages.rocm-runtime)</li>
    <li>rocmPackages.hsakmt.debug (rocmPackages.rocm-runtime.debug)</li>
    <li>rocmPackages.llvm.bintools</li>
    <li>rocmPackages.llvm.clang-unwrapped</li>
    <li>rocmPackages.llvm.clang-unwrapped.debug</li>
    <li>rocmPackages.llvm.clang-unwrapped.dev</li>
    <li>rocmPackages.llvm.clang-unwrapped.lib</li>
    <li>rocmPackages.llvm.clang-unwrapped.python</li>
    <li>rocmPackages.llvm.compiler-rt (rocmPackages.llvm.compiler-rt-libc)</li>
    <li>rocmPackages.llvm.compiler-rt.dev (rocmPackages.llvm.compiler-rt-libc.dev)</li>
    <li>rocmPackages.llvm.libcxx</li>
    <li>rocmPackages.llvm.libcxx.dev</li>
    <li>rocmPackages.llvm.lld</li>
    <li>rocmPackages.llvm.lld.dev</li>
    <li>rocmPackages.llvm.lld.lib</li>
    <li>rocmPackages.llvm.llvm</li>
    <li>rocmPackages.llvm.llvm.dev</li>
    <li>rocmPackages.llvm.llvm.lib</li>
    <li>rocmPackages.llvm.llvm.python</li>
    <li>rocmPackages.openmp (rocmPackages.llvm.openmp)</li>
    <li>rocmPackages.openmp.dev (rocmPackages.llvm.openmp.dev)</li>
    <li>rocmPackages.stdenv (rocmPackages.rocmClangStdenv)</li>
    <li>rocmPackages.migraphx</li>
    <li>rocmPackages.miopen (rocmPackages.miopen-hip)</li>
    <li>rocmPackages.mivisionx (rocmPackages.mivisionx-hip)</li>
    <li>rocmPackages.mivisionx-cpu</li>
    <li>rocmPackages.mpi (rocmPackages.openmpi)</li>
    <li>rocmPackages.mpi.dev (rocmPackages.openmpi.dev)</li>
    <li>rocmPackages.mpi.man (rocmPackages.openmpi.man)</li>
    <li>rocmPackages.mscclpp</li>
    <li>rocmPackages.rccl</li>
    <li>rocmPackages.rdc</li>
    <li>rocmPackages.rdc.doc</li>
    <li>rocmPackages.rocalution</li>
    <li>rocmPackages.rocblas</li>
    <li>rocmPackages.rocblas.benchmark</li>
    <li>rocmPackages.rocblas.test</li>
    <li>rocmPackages.rocdbgapi</li>
    <li>rocmPackages.rocdbgapi.doc</li>
    <li>rocmPackages.rocfft</li>
    <li>rocmPackages.rocgdb</li>
    <li>rocmPackages.rocm-bandwidth-test</li>
    <li>rocmPackages.rocm-cmake</li>
    <li>rocmPackages.rocm-comgr</li>
    <li>rocmPackages.rocm-core</li>
    <li>rocmPackages.rocm-device-libs</li>
    <li>rocmPackages.rocm-smi</li>
    <li>rocmPackages.rocm-tests</li>
    <li>rocmPackages.rocminfo</li>
    <li>rocmPackages.rocmlir (rocmPackages.rocmlir-rock)</li>
    <li>rocmPackages.rocprim</li>
    <li>rocmPackages.rocprofiler</li>
    <li>rocmPackages.rocprofiler-register</li>
    <li>rocmPackages.rocr-debug-agent</li>
    <li>rocmPackages.rocrand</li>
    <li>rocmPackages.rocsolver</li>
    <li>rocmPackages.rocsparse</li>
    <li>rocmPackages.rocthrust</li>
    <li>rocmPackages.roctracer</li>
    <li>rocmPackages.rocwmma</li>
    <li>rocmPackages.rpp (rocmPackages.rpp-hip)</li>
    <li>rocmPackages.rpp-cpu</li>
    <li>rocmPackages.tensile</li>
    <li>rocmPackages.tensile.dist</li>
    <li>stable-diffusion-cpp-rocm</li>
    <li>zluda</li>
  </ul>
</details>

Per commit rebuilds:


- 2ecb8730de10 (+0    -0    ~0   ) makeImpureTest: preserve NIXPKGS_ALLOW_UNFREE env var in sudo call
- 209b9e215cee (+0    -0    ~91  ) rocmPackages: 6.4.3 -> 7.0.1
- 08acac06160a (+0    -0    ~89  ) rocmPackages: 7.0.1 -> 7.0.2
- 42390a6e678f (+0    -0    ~1   ) rocmPackages.rdc: fix upstream CMake issues with build types and libcap include
- bed4cc638b72 (+0    -0    ~3   ) rocmPackages.migraphx: fix unicode minus sign breaking CArrays embed
- 50ef4960c7d6 (+0    -0    ~0   ) rocmPackages.clr: add HIP impure tests for generic-arch and isa-compat
- 144aaaf5fd1c (+0    -0    ~0   ) zluda: use finalAttrs
- e8a3c343fedd (+0    -0    ~1   ) zluda: 4-unstable-2025-01-28 -> 6-preview.45
- 5550c52f9f60 (+0    -0    ~3   ) rocmPackages.aotriton: 0.10b -> 0.11.1b
- a19b6609855f (+0    -0    ~1   ) rocmPackages.hiprt: 2.5.a21e075.3 -> 3.0.3.a1525e7
- f5a7a6ae53c2 (+0    -0    ~0   ) blender-hip: manual backport of hiprt 3 support
- 3d8704b9a287 (+1    -0    ~1   ) opensplat: add ROCm support


## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

LunNova committed:
rocmPackages: 6.4.3 -> 7.0.1

GZGavinZhao commented:
I'm currently targeting ROCm 7.0.x, uploading patches to the corresponding `solus-rocm-7.0.x` in the next few days.

GZGavinZhao commented:
Now that projects are getting unified into `ROCm/rocm-systems` and `ROCm/rocm-libraries` the patching experience should be much more unified, instead of jumping around 7 different folders :sweat_smile:

Redhawk18 commented:
Trying to build on my computer and it fails, maybe I'm not running this correctly?

```
nixpkgs on  lunnova/rocm-7-inplace 
❯ gh pr checkout 469378
From github.com:NixOS/nixpkgs
 * branch                      refs/pull/469378/head -> FETCH_HEAD
Already up to date.

nixpkgs on  lunnova/rocm-7-inplace took 2s 
❯ nix-build -A rocmPackages
error:
       … while evaluating the attribute 'clang-ocl'
         at /home/redhawk/code/nixpkgs/pkgs/development/rocm-modules/default.nix:286:7:
          285|
          286|       clang-ocl = throw ''
             |       ^
          287|         'clang-ocl' has been deprecated upstream. Use ROCm's clang directly.

       … while calling the 'throw' builtin
         at /home/redhawk/code/nixpkgs/pkgs/development/rocm-modules/default.nix:286:19:
          285|
          286|       clang-ocl = throw ''
             |                   ^
          287|         'clang-ocl' has been deprecated upstream. Use ROCm's clang directly.

       error: 'clang-ocl' has been deprecated upstream. Use ROCm's clang directly.


```

fidgetingbits commented:
The build for me worked in general, though haven't done much testing yet. That said, looks like amdsmi fails atm because it's trying to apply an old patch that's been integrated already:

```
       Output paths:
         /nix/store/s9v4j3xrivrvk62mi5gjc2brd1dpl02m-amdsmi-7.0.1
       Last 15 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/p5xfhr0zmh6xy3203nlh7mavww64i5j9-source
       > source root is source
       > Running phase: patchPhase
       > applying patch /nix/store/1j2bp2i0q9wigk34f1fl76z3wqla8fqd-esmi-to-tag-4.2.patch
       > patching file CMakeLists.txt
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 1 out of 1 hunk ignored -- saving rejects to file CMakeLists.txt.rej
       > patching file src/amd_smi/amd_smi.cc
       > Reversed (or previously applied) patch detected!  Assume -R? [n]
       > Apply anyway? [n]
       > Skipping patch.
       > 1 out of 1 hunk ignored -- saving rejects to file src/amd_smi/amd_smi.cc.rej
       For full logs, run:
         nix log /nix/store/jkdrscw76an36gm7gg1hlnly7fwb99jd-amdsmi-7.0.1.drv
```

The updated amdsmi package is still adding [this patch](https://github.com/ROCm/amdsmi/commit/49aa2af045a4bc688e6f3ee0545f12afc45c1efe.patch), but I've confirmed the source for [7.0.x](https://github.com/ROCm/amdsmi/blob/release/rocm-rel-7.0.2.1/src/amd_smi/amd_smi.cc#L6021) has the patch applied, so should be able to remove it I think.

joostn commented on pkgs/development/python-modules/torch/source/default.nix:
```diff
@@ -307,6 +309,16 @@ buildPythonPackage.override { inherit stdenv; } rec {
     ./fix-cmake-cuda-toolkit.patch
     ./nvtx3-hpp-path-fix.patch
   ]
+  ++ lib.optionals (rocmSupport && (lib.versionAtLeast rocmPackages.clr.version "7.0")) [
```
I also had to update libaotriton to 0.11b in order to get it to work, with 0.10b I got segfaults when using pytorch. I don't understand much of all this so that could well be due to my misunderstanding.

FWIW, to get libaotriton 0.11b to build, I had to add this:
```
                  env = oldAttrs.env // {
                    AOTRITON_CI_SUPPLIED_SHA1 = "0000000000000000000000000000000000000000";
                  };
```
to resolve a build error 'write_image_signature.py: error: the following arguments are required: output_file `

LunNova replied:
If you're able to test the current state of this PR's torchWithRocm+aotriton0.10 without other python overlays I'd appreciate it!

LunNova replied:
Rolled the aotriton bump into this PR!

LunNova committed:
rocmPackages: 7.0.1 -> 7.0.2

LunNova committed:
rocmPackages.hiprt: 2.5.a21e075.3 -> 3.0.3.a1525e7

PR to drop encryption support was accepted upstream so we can simplify the build slightly.
Changes from 2.5:

- RDNA 4 support (compressed BVH8, triangle packets, intersectable instances, OBB)
- new generic BVH2 import API
- per-context logger
- removed kernel encryption
- various bug fixes for RDNA 2
- ROCm 7 support

LunNova committed:
rocmPackages.rdc: fix upstream CMake issues with build types and libcap include

Applies:
- fix(rdc): Fix CXXFLAGS clobbering and incompatibility with non-x64 architecture ROCm/rocm-systems#2423
- fix(rdc): use pkg-config to find libcap ROCm/rocm-systems#2424

LunNova committed:
rocmPackages.migraphx: fix unicode minus sign breaking CArrays embed

A comment in src/targets/gpu/kernels/include/migraphx/kernels/bit.hpp
contains a Unicode MINUS SIGN instead of ASCII hyphen:
  // popcount(~(x | −x))

When EMBED_USE=CArrays, the cmake Embed.cmake mechanism reads source
files as hex and generates char[] arrays. The UTF-8 encoding of U+2212
(0xe2 0x88 0x92) causes a compile error:
  constant expression evaluates to 226 which cannot be narrowed
         to type 'char' [-Wc++11-narrowing]

Redhawk18 commented:
Is there anything that needs testing? Or you want tested?

LunNova commented:
`bash $(nix-build -A rocmPackages.clr.passthru.impureTests.isa-compat)` on that branch might do it but `name = "hip-example-isa-compat";` needs fixed to be `isa-compat`

LunNova commented:
```
$ nix run .#pkgsRocm.llama-cpp -- -hf ggml-org/functiongemma-270m-it-GGUF -c 0 -fa 1 -p "hello" -ub 1 -b 1
ggml_cuda_init: GGML_CUDA_FORCE_MMQ:    no
ggml_cuda_init: GGML_CUDA_FORCE_CUBLAS: no
ggml_cuda_init: found 1 ROCm devices:
  Device 0: AMD Ryzen 9 7950X 16-Core Processor, gfx1036 (0x1036), VMM: no, Wave Size: 32
common_download_file_single_online: using cached file: ~/.cache/llama.cpp/ggml-org_functiongemma-270m-it-GGUF_functiongemma-270m-it-bf16.gguf

Loading model...  


▄▄ ▄▄
██ ██
██ ██  ▀▀█▄ ███▄███▄  ▀▀█▄    ▄████ ████▄ ████▄
██ ██ ▄█▀██ ██ ██ ██ ▄█▀██    ██    ██ ██ ██ ██
██ ██ ▀█▄██ ██ ██ ██ ▀█▄██ ██ ▀████ ████▀ ████▀
                                    ██    ██
                                    ▀▀    ▀▀

build      : b7445-4b2a477
model      : ggml-org/functiongemma-270m-it-GGUF
modalities : text

available commands:
  /exit or Ctrl+C     stop or exit
  /regen              regenerate the last response
  /clear              clear the chat history
  /read               add a text file


> hello

Hello! How can I help you today?

[ Prompt: 55.4 t/s | Generation: 20.1 t/s ]
```

GZGavinZhao committed:
rocmPackages.clr: add HIP impure tests for generic-arch and isa-compat

Co-Authored-By: Luna Nova <git@lunnova.dev>

LunNova committed:
makeImpureTest: preserve NIXPKGS_ALLOW_UNFREE env var in sudo call

LunNova committed:
zluda: use finalAttrs

LunNova committed:
zluda: 4-unstable-2025-01-28 -> 6-preview.45

Bumping to a new unstable tag for ROCm 7.x support

LunNova committed:
rocmPackages.aotriton: 0.10b -> 0.11.1b

Diff: https://github.com/ROCm/aotriton/compare/0.10b...0.11.1b

LunNova committed:
blender-hip: manual backport of hiprt 3 support

Minimal backport that applies on current blender of
https://projects.blender.org/blender/blender/pulls/144889

LunNova committed:
opensplat: add ROCm support

joostn commented:
Works for me out of the box, 7900 XTX. Thanks!

amarshall commented:
> Does blender use rocm by default, if rocmSupport is enabled?

@Redhawk18 See https://github.com/NixOS/nixpkgs/pull/463064, which removes blender-hip in favor of rocmSupport.

amarshall reviewed (approved):
Good for blender. (Well, HIP RT seems to not be working anymore maybe, but same before this PR.)

Flakebi reviewed (commented):
Thanks, looks good. A few nits inline
